ci/README.md

```diff
@@ -0,0 +1,42 @@
+Test Infrastructure
+===================
+
+The ci directory contains scripts that aid in the testing of nix both
+in our continuous integration environment (Travis CI) but also for
+developers working locally.
+
+Nix interfaces very directly with the underlying platform (usually via
+libc) and changes need to be tested on a large number of platforms to
+avoid problems.
+
+Running Tests For Host Architecture
+-----------------------------------
+
+Running the tests for one's host architecture can be done by simply
+doing the following:
+
+    $ cargo test
+
+Running Tests Against All Architectures/Versions
+------------------------------------------------
+
+Testing for other architectures is more involved.  Currently,
+developers may run tests against several architectures and versions of
+rust by running the `ci/run-all.sh` script.  This scripts requires
+that docker be set up.  This will take some time:
+
+    $ ci/run-all.sh
+
+The list of versions and architectures tested by this can be
+determined by looking at the contents of the script.  The docker image
+used is [posborne/rust-cross][posborne/rust-cross].
+
+Running Test for Specific Architectures/Versions
+------------------------------------------------
+
+Suppose we have a failing test with Rust 1.6/1.7 on the raspberry pi.  In
+that case, we can run the following:
+
+    $ RUST_VERSIONS="1.6.0 1.7.0" RUST_TARGETS="arm-unknown-linux-gnueabihf" ci/run-all.sh
```

Comment:
No, I'll need to update this doc.  There is an equivalent, but it would be something like this:

```
DOCKER_IMAGE=posborne/rust-cross:arm RUST_VERSION="1.7.0" RUST_TARGET="arm-unknown-linux-gnueabihf" ci/run-docker.sh
```

Right now the containers only container 1.7.0.  Adding other versions is not hard but it also starts to blow up the image size.  I figured stable was the most valuable version to test against if you are going to test against anything.